packages/twenty-server/src/engine/api/graphql/graphql-query-runner/parsers/graphql-query-filter/graphql-query-filter-field.parser.ts

```diff
@@ -0,0 +1,111 @@
+import { isCompositeFieldMetadataType } from 'src/engine/metadata-modules/field-metadata/utils/is-composite-field-metadata-type.util';
+import { CompositeFieldMetadataType } from 'src/engine/metadata-modules/workspace-migration/factories/composite-column-action.factory';
+import { capitalize } from 'src/utils/capitalize';
+
+import { GraphqlQueryFilterConditionParser } from './graphql-query-filter-condition.parser';
+import { GraphqlQueryFilterOperatorParser } from './graphql-query-filter-operator.parser';
+
+export class GraphqlQueryFilterFieldParser {
+  private fieldMetadataMap: FieldMetadataMap;
+  private operatorParser: GraphqlQueryFilterOperatorParser;
+
+  constructor(fieldMetadataMap: FieldMetadataMap) {
+    this.fieldMetadataMap = fieldMetadataMap;
+    this.operatorParser = new GraphqlQueryFilterOperatorParser();
+  }
+
+  /**   *
+   * @param key - The key of the filter field.
+   * @param value - The value of the filter field.
+   * @param isNegated - Indicates whether the filter should be negated.
+   * @returns The `FindOptionsWhere` object representing the parsed filter field.
+   */
+  public parse(
+    key: string,
+    value: any,
+    isNegated: boolean,
+  ): FindOptionsWhere<ObjectLiteral> {
+    const fieldMetadata = this.fieldMetadataMap[key];
+
+    if (!fieldMetadata) {
+      return {
+        [key]: (value: RecordFilter, isNegated: boolean) => {
+          const conditionParser = new GraphqlQueryFilterConditionParser(
+            this.fieldMetadataMap,
+          );
+
+          return conditionParser.parse(value, isNegated);
+        },
+      };
+    }
+
+    if (isCompositeFieldMetadataType(fieldMetadata.type)) {
+      return this.handleCompositeFieldForFilter(
+        fieldMetadata,
+        value,
+        isNegated,
+      );
+    }
+
+    if (typeof value === 'object' && value !== null) {
+      const parsedValue = this.operatorParser.parseOperator(value, isNegated);
+
+      return { [key]: parsedValue };
+    }
+
+    return { [key]: isNegated ? Not(value) : value };
+  }
+
+  private handleCompositeFieldForFilter(
+    fieldMetadata: FieldMetadataInterface,
+    fieldValue: any,
+    isNegated: boolean,
+  ): FindOptionsWhere<ObjectLiteral> {
+    const compositeType = compositeTypeDefinitions.get(
+      fieldMetadata.type as CompositeFieldMetadataType,
+    );
+
+    if (!compositeType) {
+      throw new Error(
+        `Composite type definition not found for type: ${fieldMetadata.type}`,
+      );
+    }
+
+    return Object.entries(fieldValue).reduce(
+      (result, [subFieldKey, subFieldValue]) => {
+        const subFieldMetadata = compositeType.properties.find(
+          (property) => property.name === subFieldKey,
+        );
+
+        if (!subFieldMetadata) {
+          throw new Error(
+            `Sub field metadata not found for composite type: ${fieldMetadata.type}`,
+          );
+        }
+
+        const fullFieldName = `${fieldMetadata.name}${capitalize(subFieldKey)}`;
```

Comment:
I feel it would be great to have a helper in metadata for that : generateCompositeDatabaseColumnName

btw to me:
```
field: {
 name: 'myLinks',
 type: 'LINKS'
 property: [
   {
      name: 'primaryLinkUrl',
      type: 'TEXT'
   }
]
```

Reply:
go generateCompositeDatabaseColumnName(field)  = `${field.name}${property.name}`

And I would even do that for scalars allowing property.name to be null